logging-modules/log4j2-programmatic-configuration/set-configuration-factory/src/main/java/com/baeldung/log4j2/config/CustomConfigurationFactory.java

```diff
@@ -0,0 +1,96 @@
+package com.baeldung.log4j2.config;
+
+import java.io.IOException;
+import java.net.URI;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.ConfigurationFactory;
+import org.apache.logging.log4j.core.config.ConfigurationSource;
+import org.apache.logging.log4j.core.config.builder.api.AppenderComponentBuilder;
+import org.apache.logging.log4j.core.config.builder.api.ComponentBuilder;
+import org.apache.logging.log4j.core.config.builder.api.ConfigurationBuilder;
+import org.apache.logging.log4j.core.config.builder.api.FilterComponentBuilder;
+import org.apache.logging.log4j.core.config.builder.api.LayoutComponentBuilder;
+import org.apache.logging.log4j.core.config.builder.api.LoggerComponentBuilder;
+import org.apache.logging.log4j.core.config.builder.api.RootLoggerComponentBuilder;
+import org.apache.logging.log4j.core.config.builder.impl.BuiltConfiguration;
+
+public class CustomConfigurationFactory extends ConfigurationFactory {
+
+    static Configuration createConfiguration(final String name, ConfigurationBuilder<BuiltConfiguration> builder) {
+        AppenderComponentBuilder console = builder.newAppender("Stdout", "Console");
+        LayoutComponentBuilder layout = builder.newLayout("PatternLayout")
+            .addAttribute("pattern", "%d [%t] %-5level: %msg%n%throwable");
+        console.add(layout);
+        FilterComponentBuilder filter = builder.newFilter("MarkerFilter", Filter.Result.ACCEPT, Filter.Result.DENY);
+        filter.addAttribute("marker", "FLOW");
+        console.add(filter);
+        builder.add(console);
+
+        ComponentBuilder triggeringPolicy = builder.newComponent("Policies")
+            .addComponent(builder.newComponent("CronTriggeringPolicy")
+                .addAttribute("schedule", "0 0 0 * * ?"))
+            .addComponent(builder.newComponent("SizeBasedTriggeringPolicy")
+                .addAttribute("size", "100M"));
+        AppenderComponentBuilder rollingFile = builder.newAppender("rolling", "RollingFile");
+        rollingFile.addAttribute("fileName", "target/rolling.log");
+        rollingFile.addAttribute("filePattern", "target/archive/rolling-%d{MM-dd-yy}.log.gz");
+        rollingFile.add(layout);
+        rollingFile.addComponent(triggeringPolicy);
+        builder.add(rollingFile);
+
+        /*AppenderComponentBuilder syslog = builder.newAppender("syslogAppender", "Syslog");
```

Comment:
Let's clean this up. Either leave it in, take it out, or move it to another example. Same with the other comments below.